tests/e2e/performance/test_pvc_creation_performance.py

```diff
@@ -26,6 +48,92 @@ class TestPVCCreationPerformance(E2ETest):
+        created in the backend.
+
+        Args:
+            pool_name (str): the name of the pool to look into
+            resource (str): the name of the resource (block / filesystem)
+
+        Return:
+            int : then number of images / sub volumes
+
+        """
+        ceph = CephCluster()  # ceph cluster object to be used in the test
+        if resource == constants.CEPHBLOCKPOOL:
+            ceph_cmd = f"rbd ls --pool={pool_name}"
+        elif resource == constants.CEPHFILESYSTEM:
+            ceph_cmd = f"ceph fs subvolume ls {pool_name} --group_name=csi"
+        results = ceph.toolbox.exec_cmd_on_pod(ceph_cmd, out_yaml_format=False)
+        if resource == constants.CEPHBLOCKPOOL:
+            results = results.split("\n")
+            results.pop()  # remove last empty element from the end of the list.
+        elif resource == constants.CEPHFILESYSTEM:
+            data = results.split("\n")
+            results = []
+            for line in data:
+                if "name" in line:
+                    results.append(line.split()[-1])
+
+        log.debug(f"The PV images that found are : {results}")
+        log.info(f"Number of PV images that found is : {len(results)}")
+        return len(results)
+
+    def pvc_count_validation(
+        self, storgeclass, namespace, test_state="start", **kwargs
+    ):
+        """
+        Validate the number of PVCs -> PVs -> backend images are "as expected"
+
+        Args:
+            storageclass (str): storageclass of the PVCs to check
+            namespace (str): the namespace where the PVCs created
+            test_state (str): state of this check, start / end operation.
+               in case of start operation, collect the numbers of the images,
+               PVCs, PVs and return them.
+               in case of end operation, validate that the numbers of images is
+               as expected.
+            kwargs (dict): dictionary with numbers of PVCs, PVs, Images and the
+               number of PVCs need to be created.
+
+        Returns:
+            dict / bool: in case of state="start", return dict with numbers of :
+               PVCs, PVs, Images.
+               in case of state="end", return True if the numbers are as expected,
+               and false otherwise.
+
+        """
+
+        results = {"img": 0, "pvc": 0, "pv": 0}
+
+        pool_name = get_storage_pool_name(self.interface, namespace)
+        log.debug(f"The pool name is {pool_name}")
+
+        results["img"] = self.get_rbd_or_fs_image_count(
+            pool_name=pool_name, resource=self.interface
+        )
+        results["pvc"] = len(get_all_pvcs_in_storageclass(storgeclass))
+        results["pv"] = len(get_all_pvs_in_storageclass(storgeclass))
+        log.debug(f"{results['pv']} PVC was created !")
+        log.debug(f"{results['pvc']} PV was created !")
+        log.debug(f"Number of backend images is : {results['img']}")
+
+        if test_state == "start":
```

Comment:
I'm not sure if we need to take a different approach to start and end. at any given point of time, these should match and we should assert otherwise. we can check before and after the execution of tests, but the count should always match. wdyt?